pkg/alertmanager/alertstore/store.go

```diff
@@ -20,3 +34,59 @@ type AlertStore interface {
 	// DeleteAlertConfig deletes the alertmanager configuration for an user.
 	DeleteAlertConfig(ctx context.Context, user string) error
 }
+
+// NewLegacyAlertStore returns a new alertmanager storage backend poller and store
+func NewLegacyAlertStore(cfg LegacyConfig, logger log.Logger) (AlertStore, error) {
+	if cfg.Type == configdb.Name {
+		c, err := client.New(cfg.ConfigDB)
+		if err != nil {
+			return nil, err
+		}
+		return configdb.NewStore(c), nil
+	}
+
+	if cfg.Type == local.Name {
+		return local.NewStore(cfg.Local)
+	}
+
+	// Create the object store client.
+	var client chunk.ObjectClient
+	var err error
+	switch cfg.Type {
+	case "azure":
+		client, err = azure.NewBlobStorage(&cfg.Azure)
+	case "gcs":
+		client, err = gcp.NewGCSObjectClient(context.Background(), cfg.GCS)
+	case "s3":
+		client, err = aws.NewS3ObjectClient(cfg.S3)
+	default:
+		return nil, fmt.Errorf("unrecognized alertmanager storage backend %v, choose one of: azure, configdb, gcs, local, s3", cfg.Type)
+	}
+	if err != nil {
+		return nil, err
+	}
+
+	return objectclient.NewAlertStore(client, logger), nil
+}
+
+// NewAlertStore returns a alertmanager store backend client based on the provided cfg.
+func NewAlertStore(ctx context.Context, cfg Config, cfgProvider bucket.TenantConfigProvider, logger log.Logger, reg prometheus.Registerer) (AlertStore, error) {
+	if cfg.Backend == configdb.Name {
+		c, err := client.New(cfg.ConfigDB)
+		if err != nil {
+			return nil, err
+		}
+		return configdb.NewStore(c), nil
+	}
+
+	if cfg.Backend == local.Name {
+		return local.NewStore(cfg.Local)
+	}
```

Comment:
**thought:** When I updated the ruler client, I didn't port over the local implementation, just the configdb. My main thought was that it is readonly and mainly used for testing and the new client will already have a filesystem/inmem implementation out of the box. I figured I'd leave it in the legacy config for the time being to avoid confusions with the `filesystem` backend. However, I'm fine keeping it since there are some desirable properties to a statically configured read-only Alertmanager/RuleStorage backend. Up to you if you think it's worth porting over to the config style or if we should deprecate it when we remove the old config in the future.

If we add it here, we should also port it over to the new ruler config backend as well.

Reply:
Good point. My point was to deprecate the old config as soon as we complete the rollout in prod (and we confirm the new config works fine). Unless we want to completely remove the `local` support, then I think the new config should support `local` too so that we can completely remove the old config in 2 releases.

Reply:
👍